File: widgets/change_theme.cpp
#include "include/widgets/change_theme.hpp"
#include "include/utils/icons.hpp"
#include "src/bindings.hpp"

namespace widgets {

ChangeTheme::ChangeTheme(io::Ctx *ctx) : Gtk::Button() {
  set_css_classes({"widget", "power", "padded", "clickable"});
  set_cursor("pointer");
  set_name("ChangeTheme");

  image.set(utils::Icons::change_theme);
  set_child(image);

  signal_clicked().connect([ctx]() { io::io_change_theme(ctx); });
}

} // namespace widgets

File: ui/top_bar/change_theme.py
from gi.repository import Gdk, Gtk
from utils.commands import Commands
from utils.context import ctx


class ChangeTheme(Gtk.Button):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        ctx.pub_sub.subscribe(self)

        self.set_css_classes(["widget", "power", "padded", "clickable"])
        self.set_cursor(Gdk.Cursor.new_from_name("pointer"))
        self.set_name("ChangeTheme")

        image = Gtk.Image.new_from_gicon(ctx.icons.change_theme)
        self.set_child(image)

        self.connect("clicked", self.on_click)

    def on_click(self, _):
        Commands.change_theme()
